Style-guide/README.md

```diff
@@ -344,12 +324,14 @@ class TestDatabase : Database {
 
 ## Comments
 
-When they are needed, use comments to explain **why** a particular piece of code does something. Comments must be kept up-to-date or deleted.
+When they are needed, use comments to explain **why** a particular piece of code does something. Comments must be kept up-to-date, or altogether deleted.
```

Comment:
I would change this to be something more like comments should be avoided at all costs, with the code being self-documenting. The only exceptions are when the code deals with a complex domain-specific API that not everyone is used to working with, or the code is complicated because of legacy or bugs reasons.

Reply:
> adding our rule about associating a TODO comment with a Jira ticket would be useful

Good recommendation. Will do.

Reply:
> I would change this to be something more like comments should be avoided at all costs, with the code being self-documenting. The only exceptions are when the code deals with a complex domain-specific API that not everyone is used to working with, or the code is complicated because of legacy or bugs reasons.

I think the **why** covers it in general, but we could add extra clarification about not commenting **what** code does and point about documenting with code instead.